File: src/org/frankversnel/nl/poortjes/rendering/component/DrawableShape.java
package org.frankversnel.nl.poortjes.rendering.component;

import org.frankversnel.nl.poortjes.GameObject;
import org.frankversnel.nl.poortjes.component.Transform;
import org.frankversnel.nl.poortjes.rendering.Renderer;
import org.frankversnel.nl.poortjes.rendering.RenderingManager;

public class DrawableShape extends Drawable {
	private String shapeId;

	public DrawableShape(GameObject gameObject, RenderingManager renderingManager, 
			String shapeId) {
		super(gameObject, renderingManager);
		
		this.shapeId = shapeId;
	}

	@Override
	public void draw(Renderer renderer) {
		Transform transform = getGameObject().getComponent(Transform.class);
		
		if(transform != null) {
			renderer.drawShape(transform, shapeId);
		}
	}

}

File: src/main/scala/org/frankversnel/poortjes/rendering/drawable_shape.scala
package org.frankversnel.poortjes.rendering

import org.frankversnel.poortjes.resource_loading._

import org.frankversnel.poortjes._

trait DrawableShape extends Drawable {
	protected val resourceLoader: ResourceLoader
	protected val shape: ResourceId

	override def draw(renderer: Renderer) {
		renderer.drawShape(this, shape)
	}
}
